libraries/AP_NavEKF3/AP_NavEKF3_Outputs.cpp

```diff
@@ -308,6 +308,7 @@ bool NavEKF3_core::getLLH(struct Location &loc) const
                 loc.lng = EKF_origin.lng;
                 loc.offset(lastKnownPositionNE.x + posOffsetNED.x,
                            lastKnownPositionNE.y + posOffsetNED.y);
+                loc.alt = EKF_origin.alt - lastKnownPositionD*100;
```

Comment:
Does lastKnownPositionD get cast to a int32 before the *100 is applied? If so then this is an avoidable loss of precision that would be avoided by:

loc.alt = EKF_origin.alt - (int32_t)(lastKnownPositionD*100.0f);

Reply:
Mmmm... I think the code is good.  I've faithfully copied any bug, at least - and I reckon we might have noticed this bug:-)

```
pbarker@fx:~$ cat test.cpp
#include <stdio.h>
#include <stdint.h>

int main(int argc, char *argv[])
{
    const int32_t EKF_origin_alt = 100;
    const float lastKnownPositionD = 0.1f;

    const int32_t loc_alt = EKF_origin_alt - lastKnownPositionD*100;

    ::fprintf(stderr, "f: %d\n", loc_alt);
}
pbarker@fx:~$ g++ test.cpp  -o test && ./test
f: 90
pbarker@fx:~$ 
```

I've added in a couple of .0 on the end of the 100 to make it clear that the bit's a floating-point-multiply.